lms/djangoapps/courseware/tests/test_about.py

```diff
@@ -567,3 +574,49 @@ def test_free_course_display(self):
         self.assertEqual(resp.status_code, 200)
         self.assertNotIn("Add free to Cart (Free)", resp.content)
         self.assertNotIn('<p class="important-dates-item-title">Price</p>', resp.content)
+
+
+class CourseAboutTestCaseCCX(SharedModuleStoreTestCase, LoginEnrollmentTestCase):
+    """
+    Test for unenrolled student tries to access ccx.
+    Note: Only CCX coach can enroll a student in CCX. In sum self-registration not allowed.
+    """
+    MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Set up course for tests
+        """
+        super(CourseAboutTestCaseCCX, cls).setUpClass()
+        cls.course = CourseFactory.create()
+
+    def setUp(self):
+        """
+        Set up tests
+        """
+        super(CourseAboutTestCaseCCX, self).setUp()
+
+        # Create ccx coach account
+        self.coach = coach = AdminFactory.create()
+        self.client.login(username=coach.username, password="test")
+
+        # assign role to coach
+        role = CourseCcxCoachRole(self.course.id)
+        role.add_users(coach)
+
+    def test_redirect_to_dashboard_unenrolled_ccx(self):
+        """
+        Assert that when unenrol user tries to access ccx do not allow him self-register.
```

Comment:
Typo and grammar fix. I think you mean: "When an unenrolled user tries to access CCX, do not allow the user to self-register."